urlchecker.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# url testing
+# source urlchecker.sh
+main_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+excluded_directory="*/node_modules/*"
+all_urls=()
+all_files=()
+match_file()
+{
+  for (( j=0; j<=${#all_urls[@]}; j++ )); do
+    if [ ${all_urls[$j]} = ${1} ]
+      then
+        in_files+=("${all_files[$j]}")
+       fi
```

Comment:
nit: align the `fi` with the `if`
```suggestion
    fi
```